### PR TITLE
Feature/authVC

### PR DESCRIPTION
**Pull request by DmitrySuprun**
https://user-images.githubusercontent.com/95674666/199352740-a4ad4ab3-a42c-498a-b1f0-acf6edf23397.mov

### CODE REVIEW

File: VK/VK/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        addObserver()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        removeObserver()
+    }
+
```
Comment (AlexanderKokh) [inline VK/VK/LoginViewController.swift:30]:
```suggestion
 // MARK: - Public methods
```

File: VK/VK/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
+        let checkResult = checkUserData()
+        if !checkResult {
+            showLoginError()
+        }
+        return checkResult
+    }
+
+    // MARK: - Objc Private Methods
```
Comment (AlexanderKokh) [inline VK/VK/LoginViewController.swift:39]:
```suggestion
   // MARK: - Private Methods
```

File: VK/VK/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @objc private func hideKeyboardAction() {
+        backgroundScrollView?.endEditing(true)
+    }
+
+    // MARK: - IBAction
+
+    @IBAction private func loginButton(_ sender: Any) {}
+
```
Comment (AlexanderKokh) [inline VK/VK/LoginViewController.swift:61]:
```suggestion
```

File: VK/VK/LoginViewController.swift [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            self,
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    private func checkUserData() -> Bool {
+        let login = loginTextField.text
+        let password = passwordTextField.text
+        if login == "1", password == "1" {
```
Comment (AlexanderKokh) [inline VK/VK/LoginViewController.swift:100]:
```suggestion
        if login == , password == {
```
текстовки в константы

File: VK/VK/LoginViewController.swift [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        } else {
+            return false
+        }
+    }
+
+    private func showLoginError() {
+        let alertWrongPassword = UIAlertController(title: "Ошибка", message: "Неверный пароль", preferredStyle: .alert)
+        let okButton = UIAlertAction(title: "Ok", style: .default, handler: nil)
+        alertWrongPassword.addAction(okButton)
+        present(alertWrongPassword, animated: true, completion: nil)
```
Comment (AlexanderKokh) [inline VK/VK/LoginViewController.swift:111]:
```suggestion
```
alert   в extension  UIViewController

File: VK/VK/LoginViewController.swift [hunk 6]
```diff
@@ -0,0 +1,113 @@
+// LoginViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// User authentication
+final class LoginViewController: UIViewController {
+    // MARK: - Private Visual Components
```
[the remark this answers is not in the document]
Comment (DmitrySuprun) [reply VK/VK/LoginViewController.swift:8]: ok

**Review by death654321 — APPROVED** [review]

**Pull request merged**